### ISSUE TITLE
mdbook-lint init --include-all prints stale rule count (78) that matches neither the example config nor the runtime rule count

### ISSUE THREAD

**Issue by joshrotenberg** [joshrotenberg/mdbook-lint#423]
**Priority:** medium

## Current state

`run_init_command` in `crates/mdbook-lint-cli/src/main.rs` (line 1757) hardcodes:

```rust
println!("Includes documentation for all 78 available rules");
```

This string is stale and inconsistent with every other count in the project:

- The embedded `example-mdbook-lint.toml` itself documents ~79-81 distinct rule IDs in its comments (counted via `grep -oE '^# (MD|MDBOOK|CONTENT|ADR)[0-9]+' | sort -u`).
- `mdbook-lint rules --detailed` reports **96 rules available** at runtime (55 MD + 18 MDBOOK + 13 ADR [single-document] + 10 CONTENT; the remaining 4 ADR collection rules — ADR010-013 — are registered via `register_collection_rule` and do run, but are excluded from `engine.available_rules()`/the `rules` table because they operate across multiple files rather than one document).
- README.md and CLAUDE.md both state **100 total rules** (55 + 18 + 17 + 10), which is the mathematically correct total including all 17 ADR rules.
- The example config file's own top-of-file... (comment in `example-mdbook-lint.toml`) and README's configuration section both separately claim **83 rules** ("Comprehensive reference with all 83 rules documented").

So a single user running `mdbook-lint init --include-all` sees "78", while the README says "100", and the example file elsewhere implies "83" — three different numbers for overlapping claims, none of which is verifiably the count of anything the code currently computes.

## Desired state

Pick one authoritative, code-derived count and use it everywhere, ideally computed rather than hardcoded:

- `run_init_command`'s printed message should either compute the real count from `EXAMPLE_CONFIG_TOML`'s rule comments, or from `engine.available_rules().len()` plus a fixed offset for collection rules, rather than a hardcoded literal that silently drifts every time a rule is added.
- README.md's "83 rules" reference (in the Configuration section, `example-mdbook-lint.toml` link description) should match whatever count is actually documented in that file.
- Consider having a `cargo test` or CI check that asserts the hardcoded string in `main.rs` matches the actual rule count, so this doesn't drift again.

## Code example

```rust
// crates/mdbook-lint-cli/src/main.rs, run_init_command
println!("Includes documentation for all 78 available rules");
// vs. actual counts: 96 (rules --detailed), 100 (README/CLAUDE.md total), 83 (README's other claim)
```

## Implementation notes

- `crates/mdbook-lint-cli/src/main.rs` line ~1757.
- README.md line 157: `"Comprehensive reference with all 83 rules documented"`.
- Cross-reference with the ADR collection-rule display gap (ADR010-013 missing from `rules --detailed` output) if that's judged worth fixing as part of the same pass — but that's a CLI-display concern, not strictly a docs-accuracy one, so it can be split out if preferred.

**Comment by joshrotenberg** [comment]
ready: Compute the `init --include-all` rule count from the embedded example config's comments instead of the hardcoded 78; update README's "83 rules documented" to the computed 84 and guard it with a test; drop the redundant count from docs/src/configuration.md.
